site_source_files/content/blog/2021/selenium-4-observability.md

```diff
@@ -0,0 +1,64 @@
++++
+Description = "Observability in Selenium 4"
+Title = "Observability in Selenium 4"
+Date = 2021-04-26
+Author = "Puja Jagani"
+AuthorLink = "https://twitter.com/pujagani"
+tags = ["selenium", "grid"]
+categories = ["general"]
++++
+
+Selenium server is now power-packed with [Opentelemetry](https://opentelemetry.io/) to help troubleshoot issues and optimize performance. Thus, making the Selenium server observable! 
+
+### Need for Observability 
+
+Selenium server enables distributed testing. Instead of running the test locally, the tests run on a server that makes “**smart**” decisions to run them on different remote servers. Every such remote server is capable of hosting different types of browsers and browser versions. 
+
+Selenium 3 uses Hub as the coordinator. It receives new session requests and directs them to the appropriate Node. The Node is the remote end that has browser and operating system capabilities. 
+
+Selenium 4 extends this to provide users with a way to set up a full-blown distributed system. Essentially, [Selenium 4](https://www.selenium.dev/documentation/en/grid/grid_4/components_of_a_grid/) splits the Hub into different components with additional enhancements. But the Node’s role remained the same. To ensure the backward compatibility and simplicity of setting up Grid, standalone and hub-node mode are also available in the new Selenium server. 
+
+Now picture running a full-blown distributed Selenium Grid with hundreds of Nodes with different browsers and browser versions. Testing at scale in such an infrastructure will involve large volumes of requests. How can one keep track of these requests? Enter Selenium Observability! 
+
+### How does Selenium provide observability? 
```

Comment:
Please "title case" as if this were a main heading: "How Does Selenium Provide Observability"